src/inventory_client/retry_roundtripper.go

```diff
@@ -26,16 +26,16 @@ func (rrt RetryRoundTripper) RoundTrip(req *http.Request) (res *http.Response, e
 		Factor: 2,
 		Jitter: false,
 	}
-	return rrt.retry(rrt.maxTries, b, rrt.Proxied.RoundTrip, req)
+	return rrt.retry(rrt.maxRetries, b, rrt.Proxied.RoundTrip, req)
 
 }
 
-func (rrt RetryRoundTripper) retry(maxTries uint, backoff *backoff.Backoff, fn func(req *http.Request) (res *http.Response, e error), req *http.Request) (res *http.Response, err error) {
+func (rrt RetryRoundTripper) retry(maxRetries uint, backoff *backoff.Backoff, fn func(req *http.Request) (res *http.Response, e error), req *http.Request) (res *http.Response, err error) {
 	var i uint
-	for i = 1; i <= maxTries; i++ {
+	for i = 1; i <= maxRetries; i++ {
 		res, err = fn(req)
-		if err != nil {
-			if i <= maxTries {
+		if err != nil || res.StatusCode < 200 || res.StatusCode >= 300 {
```

Comment:
Since error might be nil, add the response statusCode to the log message at line 40